wscript

```diff
@@ -126,6 +134,48 @@ def _determine_sizeof_int(conf):
         execute=True,
         msg='Checking for sizeof(int)')
 
+def _determine_architecture(conf):
+    # set architecture specific flags if not set in the environment
+    arch_modes = ['-m32', '-m64']
+    cflags_forced = False
+    ldflags_forced = False
+
+    def get_forced_arch(flag):
+        try:
+            flags = os.environ[flag].split()
+        except KeyError:
+            return None
+
+        for arg in flags:
+            for arch in arch_modes:
+                if arg == arch:
+                    return arch
+
+
+    conf.start_msg('Checking architecture')
+    if platform.architecture()[0] == '64bit':
+        arch_mode = '-m64'
+        conf.end_msg('x86_64')
+    else:
+        arch_mode = '-m32'
+        conf.end_msg('x86')
+
+    conf.start_msg('User-defined CFLAGS')
+    conf.end_msg(get_forced_arch('CFLAGS'))
+
+    conf.start_msg('User-defined LDFLAGS')
+    conf.end_msg(get_forced_arch('LDFLAGS'))
+
+    if get_forced_arch('CFLAGS') is None:
+        if ('CFLAGS' not in conf.env) or \
```

Comment:
Nitpicking: `\` is discouraged now. You can use extra parenthesis to do the same kind of breaking the logic into multiple lines.